### PR TITLE
Improve reading array of values via config API

### PR DESCRIPTION
**Pull request by aashikam**
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/20857

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/GetConfig.java [hunk 1]
```diff
@@ -65,16 +65,27 @@ public static Object get(String configKey, String type) {
...
+    @SuppressWarnings("unchecked")
     private static BArray buildArrayValue(List value) {
-        return BValueCreator.createArrayValue(value.toArray(), new BArrayType(BTypes.typeAnydata));
+        Object[] convertedValues = new Object[value.size()];
+        value.forEach(convertedValue -> convertedValues[value.indexOf(convertedValue)] =
+                getConvertedValue(convertedValue));
+        return BValueCreator.createArrayValue(convertedValues, new BArrayType(BTypes.typeAnydata));
+    }
+
+    @SuppressWarnings("unchecked")
```
Comment (pubudu91) [inline stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/GetConfig.java]: Was this annotation needed?

File: stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/GetConfig.java [hunk 2]
```diff
@@ -65,16 +65,27 @@ public static Object get(String configKey, String type) {
     @SuppressWarnings("unchecked")
     private static MapValue buildMapValue(Map<String, Object> section) {
         MapValue map = new MapValueImpl<String, Object>();
-
-        section.forEach((key, val) -> {
-            if (val instanceof String || val instanceof Long || val instanceof Double || val instanceof Boolean) {
-                map.put(key, val);
-            }
-        });
+        section.forEach((key, val) -> map.put(key, getConvertedValue(val)));
```
Comment (pubudu91) [inline stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/GetConfig.java]: Shall we replace this with a normal for loop? We now avoid the use of these as much as possible and reverting the current usages back to normal loops

File: stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/GetConfig.java [hunk 3]
```diff
@@ -65,16 +65,27 @@ public static Object get(String configKey, String type) {
...
         return map;
     }
 
+    @SuppressWarnings("unchecked")
     private static BArray buildArrayValue(List value) {
-        return BValueCreator.createArrayValue(value.toArray(), new BArrayType(BTypes.typeAnydata));
+        Object[] convertedValues = new Object[value.size()];
+        value.forEach(convertedValue -> convertedValues[value.indexOf(convertedValue)] =
```
Comment (pubudu91) [inline stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/GetConfig.java]: same comment as above regarding the use of .forEach()

File: stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/GetConfig.java (hunk 2 above)
Comment (aashikam) [reply stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/GetConfig.java]: +1.

File: stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/GetConfig.java (hunk 3 above)
Comment (aashikam) [reply stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/GetConfig.java]: Corrected.

File: stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/GetConfig.java (hunk 2 above)
Comment (aashikam) [reply stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/GetConfig.java]: Corrected.

File: stdlib/config-api/src/test/resources/datafiles/default/ballerina.conf [hunk 4]
```diff
@@ -52,3 +52,11 @@ World!!!
...
 paths=["/etc", "/tmp", "/usr/lib/"]
 ports=[9090, 9091, 9092]
+
+[[foo]]
+a=1
+b=2
+
+[[foo]]
+a=3
+b=4
```
Comment (pubudu91) [inline stdlib/config-api/src/test/resources/datafiles/default/ballerina.conf]: Shall we add test cases for negative cases and multi dimensional arrays as well? See https://github.com/toml-lang/toml/tree/v0.4.0#array

Also a bit more complex case for arrays of table and a negative case as well: https://github.com/toml-lang/toml/tree/v0.4.0#array-of-tables

Comment (aashikam) [reply stdlib/config-api/src/test/resources/datafiles/default/ballerina.conf]: Added more tests.
